### PR TITLE
Let EventSource fail fast on bad schemes

### LINKED ISSUE #25692: EventSource constructor enters wrong readyState on non-same-origin urls with non-http(s) schemes
eventsource/eventsource-constructor-non-same-origin.htm expects an EventSource with an about, mailto, javascript, etc. url to end up in readyState CLOSED when it throws an error; the error is thrown but the readyState is incorrect.

Multiple cases are failing. One case currently passes by coincidence and will fail when #25687 fixes other behavior.

### PR DESCRIPTION
**Pull request by pshaughn**
EventSource went into an infinite reconnect loop in some cases where tests wanted it to go into a hard error state; this addresses the cases where that happens because the url isn't even http(s) and will thus definitely never result in an event stream.

https://github.com/web-platform-tests/wpt/issues/4311 suggests the tests might just be too picky here; the spec does use the word "may" on relevant behavior.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #25692 

- [X] There are tests for these changes

### CODE REVIEW

File: components/script/dom/eventsource.rs [hunk 1]
```diff
@@ -472,7 +483,7 @@ impl EventSource {
                 let event_source = event_source.root();
                 if event_source.ready_state.get() != ReadyState::Closed {
                     event_source.ready_state.set(ReadyState::Closed);
-                    event_source.upcast::<EventTarget>().fire_event(atom!("error"));
+            event_source.upcast::<EventTarget>().fire_event(atom!("error"));
```
Comment (CYBAI) [inline components/script/dom/eventsource.rs]: Heh, I'm surprised fmt didn't complain about this 🤔 looks like we should keep it with original indentation 👀?

Comment (pshaughn) [reply components/script/dom/eventsource.rs]: This keeps happening to me in particular! My workflow is to just type stuff in emacs without thinking about indents, then run mach fmt afterwards.

Comment (nox) [reply components/script/dom/eventsource.rs]: Are you saying that `fmt` isn't complaining about this reindented line?

**Comment by nox** [comment]
Seems ok to me, r=me if fmt is happy.

@bors-servo delegate+

**Comment by pshaughn** [comment]
hold on, I didn't fix the indentation yet

File: components/script/dom/eventsource.rs (hunk 1 above)
Comment (pshaughn) [reply components/script/dom/eventsource.rs]: Yeah, it seems to give up when deep enough in a task! macro.

**Comment by pshaughn** [comment]
@bors-servo retry
#24726

**Comment by pshaughn** [comment]
@bors-servo retry
more of the same

**Pull request merged**